x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/ranges/ranges.tsx

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiForm } from '@elastic/eui';
+
+import { autoInterval } from '../../../../../../../../src/plugins/data/common';
+import { RangeEditor } from './range_editor';
+import { OperationDefinition } from '../index';
+import { FieldBasedIndexPatternColumn } from '../column_types';
+import { updateColumnParam, changeColumn } from '../../../state_helpers';
+
+export const DEFAULT_INTERVAL = 1000;
+// TODO: find a way to read the UI settings config from here
```

Comment:
There are two ways. One is to use the prop `uiSettings` which is passed to each `ParamEditor`. The other is to start using the `kibana_react` hook inside the ParamEditor component, which we haven't really taken advantage of in Lens so far:

```
  const { services } = useKibana();
  const uiSettingMaxBars = services.uiSettings?.get(UI_SETTINGS.HISTOGRAM_MAX_BARS);
```

I prefer you using `uiSettings` directly because we already pass it to each ParamEditor.